dev/generate_protos.py

```diff
@@ -0,0 +1,203 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+cache_dir = "build/cache"
+temp_gencode_dir = "build/proto_gencode"
+
+mlflow_protos_dir = "mlflow/protos"
+test_protos_dir = "tests/protos"
+
+
+def gen_protos(proto_dir, proto_files, lang, protoc_bin, protoc_include_path, out_dir):
+    assert lang in ["python", "java"]
+
+    subprocess.check_call(
+        [
+            protoc_bin,
+            f"-I={protoc_include_path}",
+            f"-I={proto_dir}",
+            f"--{lang}_out={out_dir}",
+            *[f"{proto_dir}/{proto_file}" for proto_file in proto_files],
+        ]
+    )
+
+
+def apply_python_gencode_replacement(file_path):
+    content = file_path.read_text()
+
+    for old, new in python_gencode_replacements:
+        content = content.replace(old, new)
```

Comment:
```suggestion
        assert old in content
        content = content.replace(old, new)
```

to ensure `old` really exists.

Reply:
we can't add this assertion.

For code simplicity, we do the replacement in all proto gencode files. Some gencode files don't have the specific string to be replaced

Reply:
got it!